cmd/otel-allocator/collector/collector.go

```diff
@@ -87,20 +86,32 @@ func (k *Client) Watch(ctx context.Context, labelMap map[string]string, fn func(
 
 	go func() {
 		for {
-			watcher, err := k.k8sClient.CoreV1().Pods(ns).Watch(ctx, opts)
-			if err != nil {
-				log.Error(err, "unable to create collector pod watcher")
-				return
-			}
-			log.Info("Successfully started a collector pod watcher")
-			if msg := runWatch(ctx, k, watcher.ResultChan(), collectorMap, fn); msg != "" {
-				log.Info("Collector pod watch event stopped " + msg)
+			if !k.restartWatch(ctx, opts, collectorMap, fn) {
 				return
 			}
 		}
 	}()
 }
 
+func (k *Client) restartWatch(ctx context.Context, opts metav1.ListOptions, collectorMap map[string]*allocation.Collector, fn func(collectors map[string]*allocation.Collector)) bool {
+	log := k.log.WithValues("component", "opentelemetry-targetallocator")
```

Comment:
Can we move this to the `NewClient()` function and save the `k.log` as this one? I don't think we need to do this so many times 😅

Reply:
Thanks for pointing this out! Moved this up to `NewClient()` as you suggested!